web/packages/shared/components/MenuLoginWithActionMenu/MenuLoginWithActionMenu.tsx

```diff
@@ -0,0 +1,138 @@
+import { ReactElement, useRef, useState } from 'react';
+
+import { ButtonBorder, Flex, Menu, MenuItem } from 'design';
+import { ButtonSize } from 'design/Button';
+import * as icons from 'design/Icon';
+import {
+  LoginItem,
+  MenuInputType,
+  MenuLogin,
+} from 'shared/components/MenuLogin';
+
+/**
+ * Displays a menu button with additional action menu attached to the right of it.
+ * <MenuLoginWithActionMenu> expects two children, a LoginItem for <MenuLogin> and <MenuItem>
+ * for action memnu.
+ *
+ * @param buttonText - button text for main menu button.
+ * @param onSelect - handles select or click in main menu items.
+ * @param getLoginItems - fetches login items.
+ * @param children - action menu items.
+ * @param placeholder - text for action menu search box or static label.
+ * @param size - button icon size.
+ * @param width - button width.
+ * @param disableSearchAndFilter - instructs underlying MenuLogin component to display static
+ *                                 label text instead of default search box.
+ *
+ * @example
+ * <MenuLoginWithActionMenu
+ *   buttonText="Log In"
+ *   size="medium"
+ *   onSelect={() => alert('item selected')}
+ *   getLoginItems= {() => LoginItem[]}
+ * >
+ *   <MenuItem>Foo</MenuItem>
+ *   <MenuItem>Bar</MenuItem>
+ * </MenuLoginWithActionMenu>
+ */
+export const MenuLoginWithActionMenu = ({
+  buttonText,
+  onSelect,
+  getLoginItems,
+  children,
+  width,
+  size = 'medium',
+  placeholder,
+  disableSearchAndFilter,
+  launchExternalUrl,
+}: {
+  buttonText: string;
+  onSelect: (e: React.SyntheticEvent, login: string) => void;
+  getLoginItems: () => LoginItem[] | Promise<LoginItem[]>;
+  children: MenuItemComponent | MenuItemComponent[];
+  width?: string;
+  size?: ButtonSize;
+  placeholder?: string;
+  disableSearchAndFilter?: boolean;
+  launchExternalUrl?: boolean;
+}) => {
+  const moreButtonRef = useRef<HTMLButtonElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <Flex>
+      <MenuLogin
+        width={width}
+        inputType={MenuInputType.FILTER}
+        onSelect={onSelect}
+        textTransform="none"
+        alignButtonWidthToMenu
+        getLoginItems={getLoginItems}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right',
+        }}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'right',
+        }}
+        style={{ borderTopRightRadius: 0, borderBottomRightRadius: 0 }}
+        buttonText={buttonText}
+        disableSearchAndFilter={disableSearchAndFilter}
+        launchExternalUrl={launchExternalUrl}
+        placeholder={placeholder}
+      />
+      <ButtonBorder
+        setRef={moreButtonRef}
+        px={1}
+        size={size}
+        onClick={() => setIsOpen(true)}
+        /* css value matches with ButtonWithMenu component */
+        css={`
+          border-left: none;
+          border-top-left-radius: 0;
+          border-bottom-left-radius: 0;
+        `}
```

Comment:
i wonder if there is a way to make this a constant or be in the styles or something so we _know_ it matches, rather than hardcoding and hoping it doesn't drift